overlord/snapstate/snapstate.go

```diff
@@ -876,3 +876,37 @@ func GadgetInfo(s *state.State) (*snap.Info, error) {
 
 	return nil, state.ErrNoState
 }
+
+// InstallMany installs everything from the given list of names.
+// Note that the state must be locked by the caller.
+func InstallMany(st *state.State, names []string, userID int) ([]string, []*state.TaskSet, error) {
+	installed := make([]string, 0, len(names))
```

Comment:
why not `make` these to the given length, and then use the `[i]` to set them directly instead of append